tpu-client/src/nonblocking/tpu_client.rs

```diff
@@ -740,21 +715,15 @@ impl LeaderTpuService {
         protocol: Protocol,
         exit: Arc<AtomicBool>,
     ) -> Result<Self> {
+        let epoch_schedule = rpc_client.get_epoch_schedule().await?;
```

Comment:
practically today this is static for the cluster, but technically it can be modified. would require a feature gate. we should updated it upon entering a new epoch

Reply:
> It is confusing because in the documentation it is written `Returns the epoch schedule information from this cluster's genesis config` https://solana.com/docs/rpc/http/getepochschedule.

yeah these docs overly specify. [this issue](https://github.com/anza-xyz/agave/issues/3801) is a good example of how they are wrong. devnet had short epochs at genesis, which were later synchronized to testnet/mb durations. internal data structures were updated, but not those exposed externally. now we have a mess.

Am I right that you suggest to update this EpochSchedule every epoch? I did corresponding changes, need to test.

yep. nice! lemme know when testing is complete. probably easier to test on `solana-test-validator` configured with shorter epochs

Reply:
PRed a docs update, fyi. https://github.com/solana-foundation/solana-com/pull/585